sound/soc/intel/boards/Kconfig

```diff
@@ -478,8 +478,8 @@ config SND_SOC_INTEL_SOF_PCM512x_MACH
 	depends on I2C && ACPI
 	depends on (SND_SOC_SOF_HDA_AUDIO_CODEC && (MFD_INTEL_LPSS || COMPILE_TEST)) ||\
 		   (SND_SOC_SOF_BAYTRAIL && (X86_INTEL_LPSS || COMPILE_TEST))
+	depends on SND_HDA_CODEC_HDMI
```

Comment:
depends on SND_HDA_CODEC_HDMI if SND_SOC_SOF_HDA_AUDIO_CODEC?

Reply:
@plbossart is the if necessary even though we have depends on SND_SOC_SOF_HDA_AUDIO_CODEC just above?

Reply:
@plbossart @ranj063 Basicly with current patch we unnecessarily depend on SND_HDA_CODEC_HDMI if user just wants to build SND_SOC_SOF_BAYTRAIL. To fix this, we need to bake in the dependency to the line above, which to me gets too messy:

	depends on (SND_SOC_SOF_HDA_AUDIO_CODEC && SND_HDA_CODEC_HDMI &&\ 
                           (MFD_INTEL_LPSS || COMPILE_TEST)) ||\
		           (SND_SOC_SOF_BAYTRAIL && (X86_INTEL_LPSS || COMPILE_TEST))

... this is becoming rather unreadable, so I'd prefer to keep just the simple "depends on SND_HDA_CODEC_HDMI". Most users will compile support for multiple platforms anyways.